server/routes/post-hire-status.ts

```diff
@@ -26,23 +26,38 @@ router.post(
     const user = userId || localUserId;
     // Validate the request body
     await validate(ParticipantPostHireStatusSchema, body);
-    const { participantId } = body;
+    const { participantIds } = body;
     // Check participant exists
-    const [participant] = await getParticipantByID(participantId);
-    if (!participant) {
+    const notParticipants = [];
+    await Promise.all(
+      participantIds.map(async (id) => {
+        const [participant] = await getParticipantByID(id);
+        if (!participant) notParticipants.push(id);
+      })
+    );
+
+    if (notParticipants.length) {
       logger.error({
         action: 'post-hire-status_post',
-        message: `Participant does not exist with id ${participantId}`,
+        message: `Participant(s) do not exist with id ${JSON.stringify(notParticipants)}`,
       });
+
       return res.status(422).send('Participant does not exist. Please check participant ID');
     }
 
     // Get Cohort
-    const cohorts = await getAssignCohort({ participantId });
-    if (cohorts.length === 0) {
+    const noCohorts = [];
+    await Promise.all(
+      participantIds.map(async (id) => {
+        const cohorts = await getAssignCohort({ participantId: id });
+        if (cohorts.length === 0) noCohorts.push(id);
+      })
+    );
```

Comment:
As with participants, this can be simplified with a filter. See my other suggestion for an implementation example.